src/viam/sdk/components/movementsensor/movementsensor.cpp

```diff
@@ -0,0 +1,123 @@
+#include <viam/sdk/components/movementsensor/movementsensor.hpp>
+
+#include <google/protobuf/descriptor.h>
+
+#include <viam/api/component/movementsensor/v1/movementsensor.grpc.pb.h>
+#include <viam/api/component/movementsensor/v1/movementsensor.pb.h>
+
+#include <viam/sdk/common/utils.hpp>
+#include <viam/sdk/components/movementsensor/client.hpp>
+#include <viam/sdk/components/movementsensor/server.hpp>
+#include <viam/sdk/registry/registry.hpp>
+#include <viam/sdk/resource/resource.hpp>
+
+namespace viam {
+namespace sdk {
+
+std::shared_ptr<ResourceRegistration> MovementSensor::resource_registration() {
```

Comment:
With the new templated type, does this function still have value? Is it intrinsically useful to be able to obtain the `ResourceRegistration` for every type? Or could this be inlined into the `init` call in each `.cpp`:

```
namespace {
bool init() {
    Registry::register_resource(MovementSensor::static_api(),
                                MovementSensorRegistration::resource_registration());
    return true;
};

// NOLINTNEXTLINE(cert-err58-cpp)
const bool inited = init();
}  // namespace
```

Or, if you take my suggestion to eliminate the CRTP:

```
namespace {
bool init() {
    Registry::register_resource(
        MovementSensor::static_api(),
        make_resource_registration<MovementSensorClient, MovementSensorServer>());
    return true;
};

// NOLINTNEXTLINE(cert-err58-cpp)
const bool inited = init();
}  // namespace
```

Reply:
removed wrapper in 58685273